src/Microsoft.ML.TimeSeries/SrCnnBatchAnomalyDetection.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.ML.Data;
+using Microsoft.ML.Data.DataView;
+using Microsoft.ML.Internal.Utilities;
+using Microsoft.ML.Numeric;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.TimeSeries
+{
+    /// <summary>
+    /// The detect modes of SrCnn models.
+    /// </summary>
+    public enum SrCnnDetectMode
+    {
+        /// <summary>
+        /// In this mode, output (IsAnomaly, RawScore, Mag).
+        /// </summary>
+        AnomalyOnly = 0,
+
+        /// <summary>
+        /// In this mode, output (IsAnomaly, AnomalyScore, Mag, ExpectedValue, BoundaryUnit, UpperBoundary, LowerBoundary).
+        /// </summary>
+        AnomalyAndMargin = 1,
+
+        /// <summary>
+        /// In this mode, output (IsAnomaly, RawScore, Mag, ExpectedValue).
+        /// </summary>
+        AnomalyAndExpectedValue = 2
+    }
+
+    // TODO: SrCnn
+    internal sealed class SrCnnBatchAnomalyDetector : BatchDataViewMapperBase<float, SrCnnBatchAnomalyDetector.Batch>
+    {
+        private readonly int _batchSize;
+        private const int _minBatchSize = 12;
+        private readonly string _inputColumnName;
+
+        private class Bindings : ColumnBindingsBase
+        {
+            private readonly DataViewType _outputColumnType;
```

Comment:
>DataViewType [](start = 29, length = 12)

`VectorDataViewType`

Reply:
Sure. I am rewriting the SrCnnBatchAnomalyDetector to have the interface that we expect after actually adding SrCnn + rewriting the batch SrCnn sample